src/olympia/hero/models.py

```diff
@@ -97,16 +99,33 @@ class PrimaryHeroImage(ModelBase):
     def __str__(self):
         return f'{self.custom_image}'
 
+    @property
+    def preview_url(self):
+        (path, fn) = os.path.split(self.custom_image.path)
+        fn = fn.decode('utf-8')
+        return f'{HERO_PREVIEW_URL}{fn}'
+
     def preview_image(self):
         if self.custom_image:
             return mark_safe('<img class="prmhero-preview" src="{}" />'.format(
-                self.custom_image.url)
+                self.preview_url)
             )
         else:
             return None
     preview_image.short_description = "Image"
     preview_image.allow_tags = True
 
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        (path, fn) = os.path.split(self.custom_image.path)
```

Comment:
`os.path.split` returns a bytes object rather than a unicode string?

Reply:
Yep, I had to convert it from a bytes object to unicode when I needed it in the preview_url.